### DISCUSSION TITLE
SoftConsole debug error

### DISCUSSION THREAD

**Discussion by maitreyakale** [polarfire-soc/polarfire-soc-documentation#337] [General]
We're using SoftConsole-v2022.2-RISC-V-747 and have build the meta-polarfie-soc-bare-metal library examples successfully but when try to launch the debug session , getting the following MESSAGE in debug trace.

Error in final launch sequence
Failed to execute MI command:
-target-select remote localhost:3333
Error message from debugger back end:
Could not read registers; remote failure reply 'E0E'
Failed to execute MI command:
-target-select remote localhost:3333
Error message from debugger back end:
Could not read registers; remote failure reply 'E0E'
Could not read registers; remote failure reply 'E0E'

Please find the attached screenshot.

![softconsole_error](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/109587121/3ecf9584-775a-47b6-a070-f49cdf369559)

**Comment by nitindeshpande** [comment]
Hi @maitreyakale, if you are referring to the [polarfire-soc-bare-metal-library](https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/tree/master) repo, please note that it has been deprecated. 
Could you please try using example projects from [polarfire-soc-bare-metal-examples](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main) repo, instead?

**Reply by maitreyakale** [reply]
Thank you for your response! Now the problem is resolved, and now able to debug the program into the ICICLE Board.Currently I am focusing for SPI testing,but SPI bare metal examples such as (SPI-flash , SPI-master-slave) aren't giving any output in the UART terminal.Debugging is happening but not showing any output!

**Comment by nitindeshpande** [comment]
Hi @maitreyakale 
For the SPI loopback, I assume, you are using the reference design with 'SPI_LOOPBACK' parameter. Refere - [icicle-kit-reference-design](https://github.com/polarfire-soc/icicle-kit-reference-design)

Just to rule out any setup issues, could you please refer to the [icicle-kit-user-guide](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/icicle-kit-user-guide.md)

I also saw another UART issue you reported few weeks back, was that specific to another kit?
https://github.com/orgs/polarfire-soc/discussions/298

**Reply by maitreyakale** [reply]
Now I am using Softconsole tool version2022.2. I am trying to test each peripherals of Polarfire SoC. In order to do that I am following Polarfire SoC bare-metal examples : https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples. But when I tried to debug SPI related programs such as mpfs-spi-flash and mpfs-spi-master-slave (which contains reference design SPI LOOP BACK), no output is displaying in UART Terminal. I rechecked all setup issues. Now also not getting any result. I tried to connect a RFID(RC522) with mikrobus connector and tried to test SPI Peripherals. that is also not working. I don't know how to resolve it.

"That UART Issue was  resolved when I changed that port."

Please help me to figure out the issue while using SPI peripheral.

**Comment by nitindeshpande** [comment]
@maitreyakale we double checked that the using the SPI LOOPBACK variant of the reference design, the SoftConsole example project does work on icicle kit and we see the UART messages as well. The U54_1 prints messages on MSS UART1. So, as long as you are looking at the correct COM port, we would expect the project to work.

About the 'mpfs-spi-flash' project, as the readme suggest, this is a reference project and not tested on icicle kit. We are in process of updating this example project with actual SPI flash tested on the icicle kit. We will have this published soon.

**Comment by maitreyakale** [comment]
We put some print statement in the u54_1 and connect Icicle Board to USB-UART terminal port to PC, and we opened 4 serial terminals COM port using putty, no print is coming on any terminal.

**Comment by nitindeshpande** [comment]
Ok,
if you are trying to use printf() API, please ensure that the [MICROCHIP_STDIO_THRU_MMUARTX](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/blob/main/driver-examples/mss/mpfs-hal/mpfs-hal-simple-demo/src/boards/icicle-kit-es/platform_config/mpfs_hal_config/mss_sw_config.h#L171) is defined.

Please refer [this](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/blob/main/driver-examples/mss/mpfs-hal/mpfs-hal-simple-demo/src/application/hart0/e51.c#L86) as an example.

**Comment by maitreyakale** [comment]
We are using Soft Console IDE, but while debugging following error is occurred, sometimes it is working fine but most of the time getting error.

Debugging connection used: using Embedded flash pro6 programmer

Configuration used:

--command "set DEVICE MPFS"
--file [board/microsemi-riscv.cfg
![debug_issue_updated](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/109587121/de76c615-2881-4544-9726-874a7d4df311)



We are using Soft Console IDE, but while debugging following error is occurred, sometimes it is working fine but most of the time getting following error.

Please provide solution into this!
![softconsole_debug_error](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/109587121/1d172d9b-e030-4b82-a9d9-6c5293f7ed01)
